src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -436,14 +405,11 @@ void SyncInferRequest::set_tensor(const ov::Output<const ov::Node>& in_port, con
             // we must define desc for dynamic case
             // otherwise we got incorrect check on shape compatibility inside isCompatible
             // because lower and upper bound will be compared
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            actualDesc = actualDesc->cloneWithNewDims(tensor_desc.getLayout() == InferenceEngine::Layout::SCALAR
-                                                          ? InferenceEngine::SizeVector{1}
-                                                          : tensor_desc.getDims());
-            OPENVINO_SUPPRESS_DEPRECATED_END
+            actualDesc = actualDesc->cloneWithNewDims(
+                ov::is_scalar(mem_desc_ptr->getShape().getDims()) ? VectorDims{1} : mem_desc_ptr->getShape().getDims());
```

Comment:
ov::is_scalar(mem_desc_ptr->getShape().getDims()) is always false, since CPU plugin's memory descriptors express scalars as first rank tensors.

```suggestion
            actualDesc = actualDesc->cloneWithNewDims(
                ov::is_scalar(tensor.get_shape()) ? VectorDims{1} : tensor.get_shape());
```

Reply:
done